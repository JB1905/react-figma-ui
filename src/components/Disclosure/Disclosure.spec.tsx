import React from 'react';
import { render } from '@testing-library/react';

import { Disclosure, DisclosureTip } from '.';

describe('Disclosure', () => {
  const tips = [
    { heading: 'Heading 1', content: 'Content 1', id: 1 },
    { heading: 'Heading 2', content: 'Content 2', id: 2 },
    { heading: 'Heading 3', content: 'Content 3', id: 3 },
  ];

  it('should render Disclosure', () => {
    const { container } = render(
      <Disclosure
        tips={tips}
        render={({ heading, content, id }, index) => (
          <DisclosureTip
            heading={heading}
            content={content}
            section={index % 2 === 0}
            expanded={id === 1}
            key={id}
          />
        )}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render Disclosure with custom className values', () => {
    const { container } = render(
      <Disclosure
        tips={tips}
        className="custom-class-name"
        render={({ heading, content, id }) => (
          <DisclosureTip
            heading={heading}
            content={content}
            className="custom-class-name"
            labelProps={{ className: 'custom-class-name' }}
            contentProps={{ className: 'custom-class-name' }}
            section={id % 2 === 0}
            expanded={id === 1}
            key={id}
          />
        )}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
