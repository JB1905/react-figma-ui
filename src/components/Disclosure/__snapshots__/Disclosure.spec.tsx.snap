// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclosure should render Disclosure 1`] = `
<div>
  <ul
    class="disclosure"
  >
    <li
      class="disclosure__item disclosure--expanded"
    >
      <div
        class="disclosure__label disclosure--section"
      >
        Heading 1
      </div>
      <div
        class="disclosure__content"
      >
        Content 1
      </div>
    </li>
    <li
      class="disclosure__item"
    >
      <div
        class="disclosure__label"
      >
        Heading 2
      </div>
      <div
        class="disclosure__content"
      >
        Content 2
      </div>
    </li>
    <li
      class="disclosure__item"
    >
      <div
        class="disclosure__label disclosure--section"
      >
        Heading 3
      </div>
      <div
        class="disclosure__content"
      >
        Content 3
      </div>
    </li>
  </ul>
</div>
`;

exports[`Disclosure should render Disclosure with custom className values 1`] = `
<div>
  <ul
    class="disclosure custom-class-name"
  >
    <li
      class="disclosure__item custom-class-name disclosure--expanded"
    >
      <div
        class="disclosure__label custom-class-name"
      >
        Heading 1
      </div>
      <div
        class="disclosure__content custom-class-name"
      >
        Content 1
      </div>
    </li>
    <li
      class="disclosure__item custom-class-name"
    >
      <div
        class="disclosure__label custom-class-name disclosure--section"
      >
        Heading 2
      </div>
      <div
        class="disclosure__content custom-class-name"
      >
        Content 2
      </div>
    </li>
    <li
      class="disclosure__item custom-class-name"
    >
      <div
        class="disclosure__label custom-class-name"
      >
        Heading 3
      </div>
      <div
        class="disclosure__content custom-class-name"
      >
        Content 3
      </div>
    </li>
  </ul>
</div>
`;
